common/nodes/database-schema.xml

```diff
@@ -466,6 +466,32 @@ CREATE TABLE firmware_mapping (
 	CONSTRAINT unique_node_firmware UNIQUE (node_id, firmware_id)
 );
 
+<!-- Virtual Machines -->
+CREATE TABLE virtual_machines (
+	id INT AUTO_INCREMENT PRIMARY KEY,
+	hypervisor_id INT DEFAULT NULL,
+	node_id INT NOT NULL,
+	memory_size INT NOT NULL,
+	cpu_cores INT NOT NULL,
+	vm_delete BOOLEAN DEFAULT 0,
+	FOREIGN KEY (hypervisor_id) REFERENCES nodes(id) ON DELETE SET NULL,
```

Comment:
Thinking about it, I think you are right, when you remove a hypervisor with any virtual machines assigned to it, remove host should raise an error along with some schema level enforcement as well. This would require the user to either reassign the vm to another hypervisor (if this should be allowed without cleanup as well I'll discuss more in your other comment) or remove the virtual machines and sync them first.